src/functional/fencil_processors/codegens/gtfn/gtfn_module.py

```diff
@@ -0,0 +1,39 @@
+from typing import Sequence
+
+from eve.codegen import format_source
+from functional.fencil_processors import cpp, defs
+from functional.fencil_processors.codegens.gtfn import gtfn_backend
+from functional.iterator.ir import FencilDefinition
+
+
+def create_source_module(
+    itir: FencilDefinition,
+    parameters: Sequence[defs.ScalarParameter | defs.BufferParameter],
+    **kwargs,
+) -> defs.SourceCodeModule:
+    function = defs.Function(itir.id, parameters)
+
+    rendered_params = ", ".join(["gridtools::fn::backend::naive{}", *[p.name for p in parameters]])
+    decl_body = f"return generated::{function.name}(nullptr)({rendered_params});"
+    decl_src = cpp.render_function_declaration(function, body=decl_body)
+    stencil_src = gtfn_backend.generate(itir, grid_type=gtfn_backend._guess_grid_type(**kwargs))
```

Comment:
```suggestion
    stencil_src = gtfn_backend.generate(
        itir,
        grid_type=gtfn_backend._guess_grid_type(**kwargs),
        offset_provider=kwargs["offset_provider"],
    )
```

Reply:
This fixes the problem with the shift not working. The proper fix would need some cleanup how and where offset provider is passed.

Reply:
Fixed like this until we have a better solution. Also parametrized one of the shift tests.